zcash_client_sqlite/src/error.rs

```diff
@@ -58,7 +61,7 @@ impl error::Error for SqliteClientError {
     fn source(&self) -> Option<&(dyn error::Error + 'static)> {
         match &self {
             SqliteClientError::InvalidMemo(e) => Some(e),
-            SqliteClientError::Bech32(e) => Some(e),
+            SqliteClientError::KeyError(KeyError::Bech32Error(e)) => Some(e),
```

Comment:
Why only the `KeyError::Bech32Error` case?

Reply:
That's the only one of the existing error types that implements `error::Error`; we don't implement that for any of the other variants of SqliteClientError that we control, so I also did not implement it here.